test/integration/integration_test.cc

```diff
@@ -24,21 +28,21 @@ TEST_F(IntegrationTest, Echo) {
   EXPECT_EQ("hello", response);
 }
 
-TEST_F(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
+TEST_P(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
```

Comment:
For all the rest of these tests that are not parameterized, I'm a little confused on what exactly is happening. They are doing both IPv4 and IPv6 right? Are the IPv6 ones working because the listener is in dual bind mode and the IPv4 connection from the fake clients just gets converted to IPv6? I think this is fine but I just want to make sure I grok what is happening.

Also, on this topic, have we discussed what the final plan is for dual bind vs. explicit bind? I talked with @jamessynge about this a while back but I can't remember where we landed.